tests/integrations/raftstore/test_compact_log.rs

```diff
@@ -84,6 +84,80 @@ fn check_compacted(
     true
 }
 
+// Test the case where cache is compacted when one node down to check
+// if it goes well after the node is back online.
+// It's mainly aimmed to test the async fetch raft log mechanism.
+#[test]
+fn test_node_cache_compact_with_one_node_down() {
+    let count = 3;
+    let mut cluster = new_node_cluster(0, count);
+
+    cluster.cfg.raft_store.raft_log_gc_count_limit = 100000;
+    cluster.cfg.raft_store.raft_log_gc_threshold = 50;
+    cluster.cfg.raft_store.raft_log_gc_size_limit = ReadableSize::mb(20);
+    cluster.cfg.raft_store.raft_log_reserve_max_ticks = 2;
+    cluster.cfg.raft_store.raft_entry_cache_life_time = ReadableDuration::millis(100);
+    cluster.run();
+
+    cluster.must_put(b"k1", b"v1");
+
+    let mut before_states = HashMap::default();
+
+    for (&id, engines) in &cluster.engines {
+        must_get_equal(engines.kv.as_inner(), b"k1", b"v1");
+        let mut state: RaftApplyState =
+            get_raft_msg_or_default(engines, &keys::apply_state_key(1));
+        let state = state.take_truncated_state();
+        // compact should not start
+        assert_eq!(RAFT_INIT_LOG_INDEX, state.get_index());
+        assert_eq!(RAFT_INIT_LOG_TERM, state.get_term());
+        before_states.insert(id, state);
+    }
+
+    cluster.stop_node(1);
+
+    for i in 1..60 {
+        let k = i.to_string().into_bytes();
+        let v = k.clone();
+        cluster.must_put(&k, &v);
+    }
+
+    // wait log gc.
+    sleep_ms(500);
+
+    cluster.run_node(1).unwrap();
+
+    // limit has not reached, should not gc.
```

Comment:
How to know async fetch is triggered?

Reply:
now, check the metrics to make sure async-fetch is performed.